### PR TITLE
[FIX] website_portal: Verification of vat number in EU country

### PR DESCRIPTION
**Pull request by simongoffin**
We have problems when trying to validate our EU VAT report because some vat number are wrongly encoded in Odoo.
There are 2 problems:

1) We use vatnumber library to check the validity of the VAT number entered by the user on website. That library removes the following characters '.' and '-' in order to perform the check and thus can return True for a number like 123.123.123.13 or -123-3454-34
And those VAT number are then saved in our database but are rejected by the state as VAT number should only contains alphanumeric character
-> Solution: prevent entering vat number with other characters than alphanumeric for users that are located in EUROPE!

2) Usually VAT is prefixed by country code, however some people in europe managed to bypass the system by prefixing their VAT number with CC instead of their country code, CC is consider valid in odoo (it means Country Code) but is not a valid European country code in general, which means that those people that should have pay some tax have not. Which is a huge problem.
-> Solution: prevent the use of CC as country code by the user when entering a vatnumber only if the user country is in EUROPE!

source: https://en.wikipedia.org/wiki/VAT_identification_number

opw:772621

### CODE REVIEW

File: addons/website_portal/controllers/main.py [hunk 1]
```diff
@@ -77,6 +77,14 @@ def details_form_validate(self, data):
             if not data.get(field_name):
                 error[field_name] = 'missing'
 
+        in_europe = False
+        if data.get('country_id'):
+            europe = request.env["res.country.group"].search([('name', '=', 'Europe')])
```
Comment (wtaferner) [inline addons/website_portal/controllers/main.py]: You may want to use ```request.env.ref('base.europe')``` instead as the name can be renamed. Also it is possible that the country group has been deleted, so combining it with the search as a fallback might improve the quality of this method.

Comment (simongoffin) [reply addons/website_portal/controllers/main.py]: @wtaferner Thanks

**Comment by simongoffin** [comment]
What do you think about it @csnauwaert ?

**Comment by simongoffin** [comment]
closed with #20239

**Comment by simongoffin** [comment]
Closed with 2deb95b852208a8dbfb9640ac8947533c1ad428b

**Pull request closed without merging**